docs/troubleshooting.asciidoc

```diff
@@ -0,0 +1,89 @@
+ifdef::env-github[]
+NOTE: For the best reading experience,
+please view this documentation at https://www.elastic.co/guide/en/apm/agent/dotnet[elastic.co]
+endif::[]
+
+[[troubleshooting]]
+== Troubleshooting
+
+Something isn't quite working as expected?
+Here are some guidelines how to find out what's going wrong.
+
+As a first step, please check if your stack is compatible with the currently <<supported-technologies,supported technologies>>.
+
+Don't worry if you can't figure out what the problem is.
+Open a topic in the https://discuss.elastic.co/c/apm[APM discuss forum]
+and we will help you out.
+
+IMPORTANT: If you do so, *please attach your debug logs* so that we can analyze the problem.
+Upload the *complete* logs to a service like https://gist.github.com.
+The logs should include everything from the application startup up until the first request has been executed.
+
+[float]
+[[no-data-sent]]
+=== No data is sent to the APM Server
+
+If neither errors nor performance metrics are being sent to the APM Server,
+it's a good idea to first check your logs and look for output just as the app starts.
+
+In case you don't see anything suspicious in the agent logs (no warning or error) it's recommended to turn the log level to `Trace` for further investigation.
+
+=== Collecting agent logs
```

Comment:
Well, this was actually what triggered me to write the whole thing :)

Problem is, people sometimes struggle with collecting logs on ASP.NET classic. As you can see it's not trivial, but I feel this is very much needed in a troubleshooting scenario - otherwise they have no logs. Also, we had lots of questions on how to do this.

I totally agree this is way too complicated. By seeing this I think 1 step we can do mid/long term is to ship `IApmLogger` implementations for popular logging systems as part of the agent. With that the part where we show the `ApmLoggerBridge` (I feel that's the most complicated part) will go away and we just say something like "If you use NLog, just enable it with the code snippet below: (and we put a 1 liner there)"  - and same for Serilog, etc.

So, given that I see lots of users failing in the 1. step of the troubleshooting (collecting logs) I'd keep this and try to simplify this part later by shipping some code that they can just plug in easily. 

What do you think?

Reply:
> Problem is, people sometimes struggle with collecting logs on ASP.NET classic. As you can see it's not trivial, but I feel this is very much needed in a troubleshooting scenario - otherwise they have no logs.

> I'd keep this and try to simplify this part later by shipping some code that they can just plug in easily.

I appreciate the explanation! This makes sense now, and I agree, we should keep it.